### ISSUE TITLE
Cross browser testing

### ISSUE THREAD

**Issue by tom-sherman** [ariakit/ariakit#624]
Opened after #368 reappeared on Firefox, something that wouldn't have happened with some cross browser testing.

There was also a mention of performance testing too in https://github.com/reakit/reakit/issues/429#issue-490585539. This would be nice to have IMO but isn't as crucial as testing functionality testing in real browsers.

I've seen BrowserStack mentioned before, is this a good avenue to go down?

**Comment by diegohaz** [comment]
> I've seen BrowserStack mentioned before, is this a good avenue to go down?

Definitely! I think we have an open source account on BrowserStack. We just need to re-add their attribution in the readme.

Unfortunately, I haven’t had time to explore this.

**Comment by tom-sherman** [comment]
Ok great, I'll take a look at BrowserStack. I've also applied for a Cypress license, I'll evaluate that too - I've heard great things about Cypress in this space.

**Comment by diegohaz** [comment]
One thing I've been trying to do is concentrate test cases, stories and, in the future, website live examples in [`__examples__`](https://github.com/reakit/reakit/tree/master/packages/reakit/src/Composite/__examples__) folders. Right now they have a few tests (in a `__tests__` folder inside) and are being used for storybook. Maybe we could use them for browser testing as well.

**Comment by tom-sherman** [comment]
After some initial investigation Cypress looks much more suited to the task of functionality testing than BrowserStack, the official [GitHub action](https://github.com/cypress-io/github-action) looks great.

Here's the plan I'm currently exploring:

Create a simple dev server that will serve all of the examples in the `__examples__` folders. We could use Storybook here, but Storybook has a very long startup time which isn't ideal if we want this to run on commit for example.

Once this is setup and I have a couple of tests added as an example I'll open a PR. By the time that's merged we'll hopefully have the Cypress OSS license come through, at which point we can work on getting the tests running in a GitHub action and auto-uploaded to the Cypress dashboard.

**Comment by Grsmto** [comment]
@tom-sherman I had a go on my own project with Browserstack Automate. If you plan to migrate your Cypress setup to it, ping me if you need help.
All my setup does right now is [renderring my example.html page](https://github.com/Grsmto/simplebar/blob/master/packages/simplebar/tests/simplebar.test.func.js) on Safari/IE/Chrome/FF. 
Right now it's using [Intern](https://theintern.io/) to run the test as [you can do functional tests](https://theintern.io/docs.html#Intern/4/docs/docs%2Fwriting_tests.md/functional-tests) with it and it supports BrowserStack.

It looks like this then on BrowserStack then:
![Screenshot 2020-05-12 at 12 15 58](https://user-images.githubusercontent.com/527559/81712582-25736900-944b-11ea-8f6b-93518a3a59a4.jpg)

**Comment by tom-sherman** [comment]
Tha KS @Grsmto, this is really helpful. 

I've had less time than I'd like to work on this recently so I've unassigned myself, if anyone else wants to pick this up then feel free!
